public/services/requests/sql.ts

```diff
@@ -37,13 +37,17 @@ export class SQLService {
     dataSourceMDSId?: string,
     errorHandler?: (error: any) => void
   ) => {
-    return this.http
-      .get(`/api/observability/query/jobs/${params.queryId}/${dataSourceMDSId}`)
-      .catch((error) => {
-        console.error('fetch error: ', error.body);
-        if (errorHandler) errorHandler(error);
-        throw error;
-      });
+    let path;
+    if (dataSourceMDSId) {
+      path = `/api/observability/query/jobs/${params.queryId}/${dataSourceMDSId}`;
```

Comment:
I wonder if you could shorten this conditional to be 
``.get(`/api/observability/query/jobs/${params.queryId}/${dataSourceMDSId ?? ""}`)``